### PR TITLE
Enhancement: Always create blocks in modals from the block document input

### PR DESCRIPTION
**Pull request by pleek91**
# Description
When selecting a block in a schema form there's an "Add" button that allows you to create a new block to use. However this usually takes you away from the current page you're on and often loses the progress you've made on the current form. 

image

The ability to create the new block in a modal was added in https://github.com/PrefectHQ/prefect-ui-library/pull/1821 but it wasn't the default option because at the time we didn't have a way to handle nested blocks (where you cannot create a block without also creating a second block). https://github.com/PrefectHQ/prefect-design/pull/1093 gives us the ability to nicely open multiple modals at once. So now if a modal is used to create a block, and a second block needs to be created, that second block can also be created in a modal without creating an ugly UX.

This PR updates the "Add" button to always open in a modal. That way the user never loses their work and doesn't have to leave the current page. This will make creating work pools and block a much better user experience. 

image

### CODE REVIEW

File: src/components/BlockCreateModal.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <p-modal v-model:showModal="internalShowModal" class="block-create-modal" :class="modalClass" title="Add a new block">
+  <p-modal v-model:showModal="internalShowModal" class="block-create-modal" :class="classes" title="Add a new block">
```
Comment (znicholasbrown) [inline src/components/BlockCreateModal.vue:2]:
```suggestion
  <p-modal v-model:showModal="internalShowModal" class="block-create-modal" :class="classes" :title="title">
```

File: src/components/BlockCreateModal.vue [hunk 2]
```diff
@@ -44,13 +52,18 @@
 
   const blockType = ref(props.providedBlockType)
   const capability = ref(props.capability ?? '')
+  const formId = randomId()
```
Comment (znicholasbrown) [inline src/components/BlockCreateModal.vue:55]: What's this id on the block schema create form doing?

File: src/components/BlockCreateModal.vue (hunk 1 above)
Comment (pleek91) [reply src/components/BlockCreateModal.vue:2]: Whoops, I actually means to keep the static title and use the heading in the form itself. I'll delete the unused title computed. 👍

File: src/components/BlockCreateModal.vue (hunk 2 above)
Comment (pleek91) [reply src/components/BlockCreateModal.vue:55]: So by adding an id to the form itself the submit button (which is outside of the form in the modal's actions) can submit the form. Something I've seen collin do that makes the modal look much nicer. The submit button has `:form-id="formId"`.

Comment (znicholasbrown) [reply src/components/BlockCreateModal.vue:55]: ahhh ok that makes sense, i hadn't looked at that, thanks!

**Review by znicholasbrown — APPROVED** [review]

**Pull request merged**